lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3594,6 +3594,72 @@ class ConvertAtenIndexSelectOp : public OpConversionPattern<AtenIndexSelectOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenCumsumOp : public OpConversionPattern<AtenCumsumOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCumsumOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+
+    Location loc = op.getLoc();
+    MLIRContext *context = op->getContext();
+    Value input = adaptor.self();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    int64_t inputRank = inputType.getRank();
+    Type elementType = inputType.getElementType();
+
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    SmallVector<AffineExpr> indexExprs;
+    for (int i = 0; i < inputRank; i++) {
+      if (i == dim) {
+        indexExprs.push_back(getAffineDimExpr(i, rewriter.getContext()) -
+                             getAffineConstantExpr(1, context));
+      } else {
+        indexExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+      }
+    }
+    auto cumsumResultMap = AffineMap::get(inputRank, 0, indexExprs, context);
+    SmallVector<AffineMap> indexingMaps = {
+        rewriter.getMultiDimIdentityMap(inputRank), cumsumResultMap};
+    SmallVector<StringRef> iteratorTypes(inputRank, "parallel");
+    SmallVector<Value> resultOutputDims;
+    for (auto i = 0; i < inputRank; i++)
+      resultOutputDims.push_back(getDimOp(rewriter, loc, input, i));
+    Value cumsumTensor =
+        createZeroInitTensor(rewriter, loc, resultOutputDims, elementType);
+    // llvm::errs()<<"before\n";
+    auto cumsumResult =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, cumsumTensor.getType(), input, cumsumTensor, indexingMaps,
+                iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value currIndex = b.create<linalg::IndexOp>(loc, 0);
```

Comment:
I think `currIndex` should be `b.create<linalg::IndexOp>(loc, dim);`

Reply:
Yes, Thanks for pointing out this.